tools/install.sh

```diff
@@ -110,21 +129,109 @@ echo "Downloading the release: '${ANKAIOS_RELEASE_URL}'"
 else
     echo "Warning: 'sha512sum' not installed. Skipping checksum validation."
 fi
 
-# prefix with sudo if install dir is not writeable with current permissions
-PREFIX="sudo"
+# Prefix with sudo if install dir is not writeable with current permissions
+BIN_SUDO="sudo"
 if [ -w "${BIN_DESTINATION}" ]; then
-    PREFIX=""
+    BIN_SUDO=""
 fi
 
 echo "Extracting the binaries into install folder: '${BIN_DESTINATION}'"
-${PREFIX} tar -xvzf "${RELEASE_FILE_NAME}" -C "${BIN_DESTINATION}"
+${BIN_SUDO} tar -xvzf "${RELEASE_FILE_NAME}" -C "${BIN_DESTINATION}/"
+
+# Install systemd unit files
+if [ -d "$SERVICE_DEST" ]; then
+    SVC_SUDO="sudo"
+    if [ -w "$SERVICE_DEST" ]; then
+        SVC_SUDO=""
+    fi
+
+    if [[ "$INSTALL_TYPE" == server || "$INSTALL_TYPE" == both ]]; then
+        $SVC_SUDO tee "$FILE_ANK_SERVER_SERVICE" >/dev/null << EOF
+[Unit]
+Description=Ankaios server
+
+[Service]
+ExecStart=/usr/local/bin/ank-server $SERVER_OPT
+
+[Install]
+WantedBy=default.target
+EOF
+    echo "Start server with 'systemctl start $ANK_SERVER_SERVICE'"
+    fi
+
+    if [[ "$INSTALL_TYPE" == agent || "$INSTALL_TYPE" == both ]]; then
+        $SVC_SUDO tee "$FILE_ANK_AGENT_SERVICE" >/dev/null << EOF
+[Unit]
+Description=Ankaios agent
+
+[Service]
+ExecStart=/usr/local/bin/ank-agent $AGENT_OPT
+
+[Install]
+WantedBy=default.target
+EOF
+    echo "Start agent with 'systemctl start $ANK_AGENT_SERVICE'"
+    fi
+
+else
+    echo "$$SERVICE_DEST not found. Skipping installation of systemd unit files for Ankaios"
+fi
+
+# Write sample state startup config
+if ! [ -s "$FILE_STARTUP_STATE" ] && [[ "$INSTALL_TYPE" == server || "$INSTALL_TYPE" == both ]]; then
+    $SVC_SUDO mkdir -p "${CONFIG_DEST}"
+    $SVC_SUDO tee "$FILE_STARTUP_STATE" >/dev/null << EOF
+workloads:
+#   nginx:
+#     runtime: podman
+#     agent: agent_A
+#     restart: true
+#     updateStrategy: AT_MOST_ONCE
+#     accessRights: # 
+#       allow: []
+#       deny: []
+#     tags:
+#       - key: owner
+#         value: Ankaios team
+#     runtimeConfig: |
+#       image: docker.io/nginx:latest
+#       commandOptions: ["-p", "8081:80"]
+EOF
+    echo "Created sample startup config in $FILE_STARTUP_STATE."
+fi
+
+# Write uninstall script
+${BIN_SUDO} tee "${BIN_DESTINATION}/${BASEFILE_ANK_UNINSTALL}" >/dev/null << EOF
+#!/bin/bash
+set -x
+[ \$(id -u) -eq 0 ] || exec sudo \$0 \$@ 
+
+if command -v systemctl; then
+    if [ -s "${FILE_ANK_SERVER_SERVICE}" ]; then
+        systemctl stop "${ANK_SERVER_SERVICE}"
+        systemctl disable "${ANK_SERVER_SERVICE}"
+        systemctl daemon-reload
+    fi
+    if [ -s "${FILE_ANK_AGENT_SERVICE}" ]; then
+        systemctl stop "${ANK_AGENT_SERVICE}"
+        systemctl disable "${ANK_AGENT_SERVICE}"
+        systemctl daemon-reload
+    fi
+fi
+rm -f "${FILE_ANK_SERVER_SERVICE}" "${FILE_ANK_AGENT_SERVICE}"
+
+rm -f "${BIN_DESTINATION}"/ank{,-server,-agent}
+rm -f "${BIN_DESTINATION}/${BASEFILE_ANK_UNINSTALL}"
```

Comment:
I took that behavior from `k3s-uninstall.sh` and it make it transparent what happens in details. Yes, keeping the `/etc/ankaios` is by intention such that the user can re-use the config when re-installing Ankaios later.

Reply:
Ok, I see. But in the view of user experience I think that it is better to output to the user not every detailed command but instead communicating what the uninstall procedure does. Because if I think in the concept of different user roles we have defined for this project it is not understandable for everybody what the output shall communicate to that person, especially the '[ 0 -q 0 ]' stuff . If a developer wants to know what the script does he just opens that and can have a look to the commands or looks in the git repo.

Reply:
OK. I have added logs to the uninstall script an disabled output of every command.